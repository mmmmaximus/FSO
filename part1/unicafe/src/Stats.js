import React from 'react'

const Stats = ({text, number}) => <plaintext>{text} {number}</plaintext>

export default Stats
