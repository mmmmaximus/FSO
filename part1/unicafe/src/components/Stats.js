import React from 'react'

const Stats = ({text, number}) => {
  return (
    <tr data-testid='Stats'>
      <td>{text}</td>
      <td>{number}</td>
    </tr>
  )
}

export default Stats
