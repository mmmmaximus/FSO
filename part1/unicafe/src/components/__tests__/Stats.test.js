import { render, screen } from '@testing-library/react'
import Stats from './../Stats'
import renderer from 'react-test-renderer'

test('Stats should render on screen', () => {
  render(<Stats text='test text' number={1} />)
  const StatsElement = screen.getByText('test text')
  expect(StatsElement).toBeInTheDocument
  expect(StatsElement.textContent).toBe('test text')
})

it('matches snapshot', () => {
  const tree = renderer.create(<Stats text='test text' number={1} />).toJSON()
  expect(tree).toMatchSnapshot()
})
