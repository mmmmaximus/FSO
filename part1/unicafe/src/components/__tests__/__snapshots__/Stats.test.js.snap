// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<tr
  data-testid="Stats"
>
  <td>
    test text
  </td>
  <td>
    1
  </td>
</tr>
`;
